scripts/credstashCli.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * Created by Idan on 6/14/18
+ */
+
+const process = require('process');
+const program = require('commander');
+const _ = require('lodash');
+const CredstashApi = require('../lib/credstashApi');
+
+function getValue(table, appKey = undefined, key = undefined) {
+    this.credStashApi = new CredstashApi({table: table});
+    let storedValue = null;
+
+    if (appKey === undefined) {
+        storedValue = this.credStashApi.getAll();
+        Object.keys(storedValue).forEach(
+            function (keyName) {
+                if (typeof storedValue[keyName] !== 'function' && storedValue[keyName].charAt(0) === '{') {
+                    try {
+                        storedValue[keyName] = JSON.parse(storedValue[keyName]);
+                    } catch (e) {
+                        console.error("Error parsing JSON from CredStash. Error message: " + e.toString());
+                        console.error("Offending key: " + keyName);
+                        console.error("Value: " + storedValue[keyName]);
+                        process.exit(1);
+                    }
+                }
+            });
+    } else {
+        storedValue = JSON.parse(this.credStashApi.get(appKey));
```

Comment:
why here its enough to get and json parse, while above u had to parse each value separately and check for "not function" and "starting with brace"

Reply:
It's because the CredstashApi.get() function returns a string (which should be JSON formatted) while the CredstashApi.getAll() function returns an object whose properties contain strings that can either represent JSON objects themselves or be simple strings.
The reason I checked for "not function" and "starts with brace" above is to determine whether I should parse the property's value or not.